src/capabilities.js

```diff
@@ -299,13 +299,13 @@ export class Capabilities {
         "Only the project creator can assign the project creator's role"
       )
     }
-    const ownCapabilities = await this.getCapabilities(this.#ownDeviceId)
 
     if (roleId === LEFT_ROLE_ID) {
       if (deviceId !== this.#ownDeviceId) {
         throw new Error('Cannot assign LEFT role to another device')
       }
     } else {
+      const ownCapabilities = await this.getCapabilities(this.#ownDeviceId)
```

Comment:
While I was developing, I ran into issues with this being in its original location. I think this is a better spot for it anyway, so I kept it.